### ISSUE TITLE
WB engine cores option was written twice

### ISSUE THREAD

**Issue by fsmosca** [cutechess/cutechess#217]
![image](https://cloud.githubusercontent.com/assets/22366935/26615707/9d8dcc42-45fb-11e7-8476-a7235819754b.png)

Dirty 30april2017 can be downloaded from here.
http://www.dirtychess.com/downloads.php

I am using Cutechess GUI 0.9.4 after 1883 commits.

**Comment by alwey** [comment]
cutechess creates a spin type option named `cores` by itself when it finds an engine with feature `smp=1`. Before a game starts it sends the` cores N` command to such an engine to specify the maximum number of (high load) threads.

Dirty already has a feature `cores`, but e.g., Crafty has not. 

Dirty
```
feature smp=1
feature option="cores -spin 1 1 16"
```
Crafty 23.4
```
feature smp=1
```
For Dirty both features given by the UI and by the engine are listed. They are treated differently.
Cutechess sends `cores 1` resulting from its own spin type option (listed at the top) and
`option cores=1` from Dirty's feature which is listed after `bookmin`.

**Comment by fsmosca** [comment]
According to xboard protocol, the gui should display for those feature options.

> smp (boolean, default 0)
> If smp=1, the GUI will send the "cores" command to the engine to inform it how many CPU cores it can use. Note that sending smp=1 does not imply the engine can use more than one CPU; just that it wants to receive the "cores" command.
>
> option (string, see text below)
> This feature is used by the engine to define an option command to appear in a GUI menu, so that the user can change the corresponding setting of the engine through the GUI interactively....

https://www.gnu.org/software/xboard/engine-intf.html

So perhaps Cutechess should only build a menu option in the gui, for feature options and ignore the feature smp.

**Comment by alwey** [comment]
> So perhaps Cutechess should only build an option in the gui, for feature options and ignore building of an option in the gui for the feature smp.

I do not think this alone would be a good idea. It would break communication with engines. It is clear that `smp=1` should invoke the `cores` command as given by the specification.

Dirty is the odd one out of the xboard smp-engines on my machine, with both `featue smp=1` and spin type `feature option cores`:

crafty 23.4 and 25.2, gaviota 1.0, and hoichess 0.19.0:  
`feature smp=1`

scorpio 2.7.7:
```
feature smp=0
feature option="cores -spin 1 1 32"
```
Dirty Apr 30 2017:
```
feature smp=1
feature option="cores -spin 1 1 16"
```

**Comment by fsmosca** [comment]
HGM has posted here.
http://talkchess.com/forum/viewtopic.php?topic_view=threads&p=717523&t=64160

**Comment by alwey** [comment]
Thank you, so it seems cutechess is in order. HGM says:

> So when an engine announces a non-standard option, the GUI has no choice other than to display it in the engine-settings dialog. The Dirty 'cores' option could do anything.

Dirty is the odd one here.

**Comment by fsmosca** [comment]
If the effect of feature smp=1 and feature option=cores... is the same which I think it is, then Cutechess should just display just 1 cores label in the GUI.

The advantage of feature option is that it tells to the user the default, min and max values, whereas the smp=1 does not tell at all.

I actually think that Dirty is more complete. It shows to the gui that it can handle cores command and tells the user the limits of the cores it can take via option=cores...

**Comment by alwey** [comment]
_Dirty_ is broken. Its SMP control does not even work with a recent _xboard_ version.

**Comment by fsmosca** [comment]
For uci engines Cutechess just uses a label called,
>Threads

from 

>option name Threads type spin default 1 min 1 max 128

This is equivalent to WB
feature option="cores -spin 1 1 16"

Xboard protocol recommends that the gui should build an option for this feature option="cores, and not for feature smp=1.

One solution could be to use smp in the cutechess label. This is for WB engines that supports feature smp=1.

![image](https://cloud.githubusercontent.com/assets/22366935/26757039/baf7b718-48e2-11e7-841c-423c208b1f9c.png)
